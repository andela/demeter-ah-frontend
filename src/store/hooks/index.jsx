import { useEffect } from 'react';
import { getUser } from '../actions/signup';

export const useSetUser = ({ dispatch }) => {
  useEffect(() => {
    dispatch(getUser());
  }, []);
  return [];
};
