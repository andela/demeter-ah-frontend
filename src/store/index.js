import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import combineReducers from './reducers';
import dummyAction from './actions/Signup';

const middlewares = applyMiddleware(thunk);

const store = createStore(
  combineReducers,
  composeWithDevTools(middlewares),
);

store.dispatch(dummyAction);

export default store;
