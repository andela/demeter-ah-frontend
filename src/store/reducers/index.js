import { combineReducers } from 'redux';
import authReducer from './Auth';
import resetPassword from './resetPassword';
import changePassword from './changePassword';
import articles from './articles';
import editProfile from './editProfile';
import viewProfile from './viewProfile';
import membership from './Membership';
import viewArticle from './viewArticle';
import relatedArticles from './relatedArticles';

const reducers = combineReducers({
  auth: authReducer,
  resetPassword,
  changePassword,
  articles,
  editProfile,
  viewProfile,
  membership,
  viewArticle,
  relatedArticles
});

export default reducers;
