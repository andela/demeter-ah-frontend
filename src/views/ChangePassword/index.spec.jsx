import React from 'react';
import { shallow, mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../store';
import ReduxChangePassword, { ChangePassword } from './index';

describe('CHANGE PASSWORD COMPONENT', () => {
  it('Should render without errors', () => {
    const component = shallow(
      <Router>
        <ReduxChangePassword store={store} />
      </Router>
    );
    expect(component).toMatchSnapshot();
  });

  it('Should render along with children componenet', () => {
    const component = mount(
      <Router>
        <ReduxChangePassword store={store} />
      </Router>
    );
    expect(component).toMatchSnapshot();
    expect(component.find('button')).toHaveLength(1);
    expect(component.find('input')).toHaveLength(2);
  });

  it('Should render status view after reset success', () => {
    const component = mount(
      <Router>
        <ChangePassword
          authResponse={{ message: 'welcome' }}
          cleanUpPasswordChange={() => { }}
          isSubmitting={false}
          changePassword={() => { }}
        />
      </Router>
    );
    expect(component).toMatchSnapshot();
  });

  it('should call onChange function', () => {
    const event = {
      preventDefault() { },
      target: { value: 'frank@gmail.com', name: 'email' },
    };
    const component = mount(
      <ChangePassword
        authResponse={{ error: 'bad data' }}
        cleanUpPasswordChange={() => { }}
        isSubmitting={false}
        changePassword={() => { }}
      />
    );
    const inputTag = component.find('input').at(0);
    inputTag.simulate('change', event);
  });

  it('should call handleSubmit function', () => {
    const mockOnResetFn = jest.fn();
    const fakeEvent = { preventDefault: () => { } };
    const component = shallow(
      <ChangePassword
        authResponse={{}}
        cleanUpPasswordChange={() => { }}
        location={{
          search: '?resetToken=ed694ccc396754f158541e'
        }}
        isSubmitting={false}
        changePassword={mockOnResetFn}
      />
    );
    component.find('form').simulate('submit', fakeEvent);
    expect(mockOnResetFn.mock.calls.length).toBe(1);
  });

  it('should call toast function', () => {
    const fakeEvent = { preventDefault: () => { } };
    const component = mount(
      <ChangePassword
        authResponse={{}}
        cleanUpPasswordChange={() => { }}
        location={{
          search: '?resetToken=ed694ccc396754f158541e'
        }}
        isSubmitting={false}
        changePassword={() => { }}
      />
    );
    const input1 = component.find('input').at(0);
    const input2 = component.find('input').at(1);
    input1.simulate('change', { target: { name: 'password', value: 'Hello' } });
    input2.simulate('change', { target: { name: 'confirmPassword', value: 'Hello1' } });
    component.find('form').simulate('submit', fakeEvent);
  });

  it('should change button value from change to loading on submit', () => {
    const component = mount(
      <ChangePassword
        authResponse={{ message: 'welcome' }}
        cleanUpPasswordChange={() => { }}
        location={{
          search: '?resetToken=ed694ccc396754f158541e'
        }}
        isSubmitting
        changePassword={() => { }}
      />
    );
    expect(component.find('button').text()).toEqual('Loading...');
  });
});
