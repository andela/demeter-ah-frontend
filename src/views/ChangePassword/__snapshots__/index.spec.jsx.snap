// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CHANGE PASSWORD COMPONENT Should render along with children componenet 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Connect(ChangePassword)
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <ChangePassword
        authResponse={Object {}}
        changePassword={[Function]}
        cleanUpPasswordChange={[Function]}
        isSubmitting={false}
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
      >
        <main
          className="change-password"
        >
          <form
            autoComplete="off"
            className="lg:w-5/12 md:w-3/5 sm:w-4/5 w-full min-w-min sm:rounded-2xl rounded-none"
            onSubmit={[Function]}
          >
            <h3
              className="text-center text-white text-3xl font-semibold"
            >
              Change Password
            </h3>
            <InputForm
              classes="my-8 md:w-3/5 w-4/5 mx-auto"
              inputType="password"
              labelname="Password"
              name="password"
              onChange={[Function]}
              placeholder="password"
            >
              <div
                className="my-8 md:w-3/5 w-4/5 mx-auto"
              >
                <Label
                  labelname="Password"
                  name="password"
                >
                  <label
                    htmlFor="password"
                  >
                    Password
                  </label>
                </Label>
                <Input
                  inputType="password"
                  name="password"
                  onChange={[Function]}
                  placeholder="password"
                >
                  <input
                    className="forminput rounded py-2 bg-purple-150 undefined"
                    name="password"
                    onChange={[Function]}
                    placeholder="password"
                    required={true}
                    type="password"
                  />
                </Input>
              </div>
            </InputForm>
            <InputForm
              classes="my-8 md:w-3/5 w-4/5 mx-auto"
              inputType="password"
              labelname="Confirm Password"
              name="confirmPassword"
              onChange={[Function]}
              placeholder="confirm password"
            >
              <div
                className="my-8 md:w-3/5 w-4/5 mx-auto"
              >
                <Label
                  labelname="Confirm Password"
                  name="confirmPassword"
                >
                  <label
                    htmlFor="confirmPassword"
                  >
                    Confirm Password
                  </label>
                </Label>
                <Input
                  inputType="password"
                  name="confirmPassword"
                  onChange={[Function]}
                  placeholder="confirm password"
                >
                  <input
                    className="forminput rounded py-2 bg-purple-150 undefined"
                    name="confirmPassword"
                    onChange={[Function]}
                    placeholder="confirm password"
                    required={true}
                    type="password"
                  />
                </Input>
              </div>
            </InputForm>
            <Button
              className="submitBtn"
              isSubmit={false}
              type="submit"
            >
              <button
                className="submitBtn"
                disabled={false}
                type="submit"
              >
                Submit
              </button>
            </Button>
          </form>
        </main>
      </ChangePassword>
    </Connect(ChangePassword)>
  </Router>
</BrowserRouter>
`;

exports[`CHANGE PASSWORD COMPONENT Should render status view after reset success 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <ChangePassword
      authResponse={
        Object {
          "message": "welcome",
        }
      }
      changePassword={[Function]}
      cleanUpPasswordChange={[Function]}
      isSubmitting={false}
    >
      <main
        className="change-password"
      >
        <Status
          height="h-86"
          status="welcome"
          width="w-2/4"
        >
          <div
            className="w-2/4 h-86 container"
          >
            <h3
              className="status"
            >
              welcome
            </h3>
            <div
              className="actions"
            >
              <Link
                to="/signin"
              >
                <a
                  href="/signin"
                  onClick={[Function]}
                >
                  Go to Sign In Â  
                  <b>
                    &gt;
                  </b>
                </a>
              </Link>
            </div>
          </div>
        </Status>
      </main>
    </ChangePassword>
  </Router>
</BrowserRouter>
`;

exports[`CHANGE PASSWORD COMPONENT Should render without errors 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Connect(ChangePassword)
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />
</Router>
`;
